src/atc/functions.py

```diff
@@ -2,10 +2,47 @@
 A collection of useful pyspark snippets.
 """
 
+from atc.spark import Spark
 from pyspark.sql.functions import udf
 from pyspark.sql.types import StringType
 import uuid as _uuid
+import pyspark.sql.functions as F
+from pathlib import Path
 
 # Pyspark uuid function implemented as recommended here
 # https://stackoverflow.com/questions/49785108/spark-streaming-with-python-how-to-add-a-uuid-column/50095755
 uuid = udf(lambda: str(_uuid.uuid4()), StringType()).asNondeterministic()  # noqa
+
+def init_dbutils(spark):
+    try:
+        from pyspark.dbutils import DBUtils
+        dbutils = DBUtils(spark)
+    except ImportError:
+        import IPython
+        dbutils = IPython.get_ipython().user_ns["dbutils"]
+    return dbutils
+
+def drop_table_cascade(DataBaseName,TableName):
+
+    #Check if table exists 
+    if Spark.get()._jsparkSession.catalog().tableExists(DataBaseName, TableName):
+
+        #### GET PATH - Idea 1 ###
+        # #Collect file path
+        # file_name = spark.read.table(f"{DataBaseName}.{TableName}").select(F.input_file_name()).take(1)
+        # #Collect parent 
+        # table_path=str(Path(file_name).parent)
+        ##########################
+
+        #### GET PATH - Idea 2 ###
+        table_path=str(Spark.get().sql(f"DESCRIBE DETAIL {DataBaseName}.{TableName}").collect()[0]["location"])
+        ##########################
+
+        if table_path is None:
+            raise Exception(f"Table path is NONE.")
```

Comment:
It is generally a good idea to make up your custom Exception classes and raise those:

    class MyUnableToCascadeException(Exception):
        pass

This allows those that have to catch it to only catch your specific one. In fact pycharm complains if you try to `except Exception` saying that this is too broad (it will catch everything including misspelled variables etc.).

Reply:
I dont understand how I should do this? Lets have a talk